qemu-tests/src/main.rs

```diff
@@ -0,0 +1,75 @@
+#![no_std]
+#![no_main]
+
+use arbitrary_int::{u12, u4, u5};
+use defmt_semihosting as _;
+use qemu_tests as _;
+
+#[bitbybit::bitenum(u2, exhaustive = true)]
+#[derive(defmt::Format)]
+pub enum TestEnum {
+    A = 0b00,
+    B = 0b01,
+    C = 0b10,
+    D = 0b11,
+}
+#[bitbybit::bitfield(u32, default = 0x0, defmt_bitfields)]
+struct TestBitfields {
+    #[bits(11..=15, rw)]
+    number: u5,
+
+    #[bit(10, rw)]
+    boolean: bool,
+
+    #[bits(5..=6, rw)]
+    enumeration: TestEnum,
+}
+
+#[bitbybit::bitfield(u32, default = 0x0, defmt_fields)]
```

Comment:
btw it's also possible to have `bitfield(u5)` (meaning that the underlying type is an arbitrary-int). This allows nested bitfields.

Might be worth having a test for it?

Reply:
I actually had to add a bit of logic so that bitfield works.

While being at that, I think the crate would definitely profit from a full proper https://github.com/dtolnay/trybuild testsuite for adding features / maintenance. These tests would simplify debugging macro implementations and additions. Right now, I have to edit and examples while sprinkling printouts inside the macro crate. Trybuild would give me a proper test harness while adding new features which also verifies correct error behaviour.

But this is something for a different PR.